File: parsers/perl-archive-zip/unzip.pl
use strict;
use Archive::Zip qw(:ERROR_CODES);

my $zip = Archive::Zip->new();
my $status = $zip->read($ARGV[0]);
die 'Failed to read ZIP' if $status != AZ_OK;
$status = $zip->extractTree('', $ARGV[1]);
die 'Failed to extract ZIP' if $status != AZ_OK;

File: parsers/rust-zip/unzip.rs
use zip::read::ZipArchive;
use std::fs::File;

fn main() {
    let mut args = std::env::args().skip(1);
    let src = args.next().expect("no src in args");
    let dst = args.next().expect("no dst in args");
    let file = File::open(src).expect("failed to open input file");
    let mut archive = ZipArchive::new(file).expect("failed to read input ZIP");
    archive.extract(dst).expect("failed to extract");
}
